src/trans/gpu/CMakeLists.txt

```diff
@@ -0,0 +1,129 @@
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+# Preprocess module file containing version information
+configure_file( internal/ectrans_version_mod.F90.in internal/ectrans_version_mod.F90 )
+
+## Apply workarounds for some known compilers
+
+if(CMAKE_Fortran_COMPILER_ID MATCHES "NVHPC")
+
+  # Compile setup_trans with pinned memory to improve data movement performance.
+  ectrans_add_compile_options(
+      SOURCES external/setup_trans.F90
+      #FLAGS   "-gpu=pinned,deepcopy,fastmath,nordc")
+      FLAGS   "-gpu=pinned,fastmath")
+  # TODO: check if it is sufficient to only set "-gpu=pinned" which appends rather than overwrites
+
+endif()
+
+## Assemble sources
+
+ecbuild_list_add_pattern( LIST trans_src
+                          GLOB
+                                sharedmem/*
+                                algor/*
+                                internal/*
+                                external/*
+                                ${CMAKE_CURRENT_BINARY_DIR}/internal/ectrans_version_mod.F90
+                          QUIET
+                        )
+
+ecbuild_list_exclude_pattern( LIST trans_src            REGEX dilatation_mod.F90 )
+ecbuild_list_exclude_pattern( LIST trans_src            REGEX cuda_gemm_batched_mod.F90 )
+
+message( "trans_src list : ${trans_src}" )
+
+#if( NOT ${CMAKE_BUILD_TYPE_CAPS} STREQUAL DEBUG )
+  set_source_files_properties( internal/ftinv_mod.F90 PROPERTIES COMPILE_OPTIONS "-O2" )
+  ecbuild_info("warn: special compile flags ftinv_mod.F90")
+  set_source_files_properties( internal/ftdir_mod.F90 PROPERTIES COMPILE_OPTIONS "-O2" )
+  ecbuild_info("warn: special compile flags ftdir_mod.F90")
+#endif()
+
+# Filter source list according to available GPU runtime
+if( HAVE_HIP )
+  set( GPU_RUNTIME "HIP" )
+  ectrans_declare_hip_sources( SOURCES_GLOB
+    sharedmem/*.hip.cpp
+    algor/*.hip.cpp
+    internal/*.hip.cpp
+    external/*.hip.cpp
+  )
+  ecbuild_list_exclude_pattern( LIST trans_src  REGEX \.cu$ )
+  ecbuild_list_exclude_pattern( LIST trans_src  REGEX cuda_device_mod.F90 )
+elseif( HAVE_CUDA )
+  set( GPU_RUNTIME "CUDA" )
+  set( ECTRANS_GPU_HIP_LIBRARIES CUDA::cufft CUDA::cublas nvhpcwrapnvtx CUDA::cudart )
+  ecbuild_list_exclude_pattern( LIST trans_src  REGEX \.hip\.cpp )
+  ecbuild_list_exclude_pattern( LIST trans_src  REGEX hip_device_mod.F90 )
+else()
+  ecbuild_info("warn: HIP and CUDA not found")
+endif()
+
+foreach( prec dp sp )
+  if( HAVE_${prec} )
+        set( GPU_LIBRARY_TYPE SHARED )
+        if( HAVE_GPU_STATIC )
+            set( GPU_LIBRARY_TYPE STATIC )
+        endif()
+
+        ectrans_add_library(
+          TARGET               trans_gpu_${prec}
+          TYPE                 ${GPU_LIBRARY_TYPE}
+          SOURCES              ${trans_src}
+          LINKER_LANGUAGE      Fortran
+          PUBLIC_INCLUDES      $<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/src/trans/include>
```

Comment:
This path does not exist : needs either to be updated to the include folder in the cpu folder, or we move the cpu includes up a level